### PR TITLE
fresh template syntax

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@smarterclayton @ncdc ptal.

@sspeiche this also adds the configchange trigger to the db templates that we talked about.

**Comment by bparees** [comment]
[test]

**Comment by bparees** [comment]
btw @smarterclayton, given that we have "20" in our ruby image name (for example), we'd never be able to add a 21 tag for an existing imagestream, because 21 would be a whole new docker repo and therefore need a new imagestream (at least until we do the repo aggregation thing you were talking about)

**Comment by bparees** [comment]
@ncdc i seem to recall you recently asking if i was going to update/clean up something in the templates/imagestreams....if this doesn't cover it, please remind me what it was :)

**Comment by smarterclayton** [comment]
Repo aggregation works now?  Just not refs to refs across IS.

**Comment by sspeiche** [comment]
@bparees ok sounds good, I had made some local changes https://github.com/sspeiche/origin/tree/cfgchg was going to submit PR but will hold off

File: examples/image-streams/image-streams-centos7.json [hunk 1]
```diff
@@ -111,30 +111,45 @@
...
       "metadata": {
-        "name": "mysql-55-centos7"
+        "name": "mysql"
       },
       "spec": {
-        "dockerImageRepository": "openshift/mysql-55-centos7"
+        "dockerImageRepository": "openshift/mysql-55-centos7",
+        "tags": [
+          {
+            "name":"latest"
```
Comment (ncdc) [inline examples/image-streams/image-streams-centos7.json]: This doesn't do anything

File: examples/image-streams/image-streams-centos7.json [hunk 2]
```diff
@@ -111,30 +111,45 @@
...
       "metadata": {
-        "name": "postgresql-92-centos7"
+        "name": "postgresql"
       },
       "spec": {
-        "dockerImageRepository": "openshift/postgresql-92-centos7"
+        "dockerImageRepository": "openshift/postgresql-92-centos7",
+        "tags": [
+          {
+            "name":"latest"
```
Comment (ncdc) [inline examples/image-streams/image-streams-centos7.json]: This doesn't do anything

File: examples/image-streams/image-streams-centos7.json [hunk 3]
```diff
@@ -111,30 +111,45 @@
...
       "metadata": {
-        "name": "mongodb-24-centos7"
+        "name": "mongodb"
       },
       "spec": {
-        "dockerImageRepository": "openshift/mongodb-24-centos7"
+        "dockerImageRepository": "openshift/mongodb-24-centos7",
+        "tags": [
+          {
+            "name":"latest"
```
Comment (ncdc) [inline examples/image-streams/image-streams-centos7.json]: This doesn't do anything

File: examples/image-streams/image-streams-rhel7.json [hunk 4]
```diff
@@ -90,30 +90,45 @@
...
       "metadata": {
-        "name": "mysql-55-rhel7"
+        "name": "mysql"
       },
       "spec": {
-        "dockerImageRepository": "registry.access.redhat.com/openshift3_beta/mysql-55-rhel7"
+        "dockerImageRepository": "registry.access.redhat.com/openshift3_beta/mysql-55-rhel7",
+        "tags": [
+          {
+            "name":"latest"
```
Comment (ncdc) [inline examples/image-streams/image-streams-rhel7.json]: This doesn't do anything

File: examples/image-streams/image-streams-rhel7.json [hunk 5]
```diff
@@ -90,30 +90,45 @@
...
       "metadata": {
-        "name": "postgresql-92-rhel7"
+        "name": "postgresql"
       },
       "spec": {
-        "dockerImageRepository": "registry.access.redhat.com/openshift3_beta/postgresql-92-rhel7"
+        "dockerImageRepository": "registry.access.redhat.com/openshift3_beta/postgresql-92-rhel7",
+        "tags": [
+          {
+            "name":"latest"
```
Comment (ncdc) [inline examples/image-streams/image-streams-rhel7.json]: This doesn't do anything

File: examples/image-streams/image-streams-rhel7.json [hunk 6]
```diff
@@ -90,30 +90,45 @@
...
       "metadata": {
-        "name": "mongodb-24-rhel7"
+        "name": "mongodb"
       },
       "spec": {
-        "dockerImageRepository": "registry.access.redhat.com/openshift3_beta/mongodb-24-rhel7"
+        "dockerImageRepository": "registry.access.redhat.com/openshift3_beta/mongodb-24-rhel7",
+        "tags": [
+          {
+            "name":"latest"
```
Comment (ncdc) [inline examples/image-streams/image-streams-rhel7.json]: This doesn't do anything

**Comment by bparees** [comment]
spoke w/ @ncdc, leaving the no-op latest tags in place, in the near future they're going to point to a version specific tag.

**Comment by ncdc** [comment]
ImageStream changes LGTM

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
needed to update IR names in test-cmd.sh.  [merge]

**Pull request merged**